src/ray/core_worker/core_worker.cc

```diff
@@ -368,6 +368,16 @@ CoreWorker::CoreWorker(const CoreWorkerOptions &options, const WorkerID &worker_
       options_.ref_counting_enabled ? reference_counter_ : nullptr, local_raylet_client_,
       options_.check_signals));
 
+  auto check_node_alive_fn = [this](const ClientID &node_id) {
+    auto node = gcs_client_->Nodes().Get(node_id);
```

Comment:
Is this a synchronous call to the GCS service? If so, consider running microbenchmarks before and after this change.

Reply:
Nah, just checks a cache.